src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,481 @@
+                    ++i;
+                }
+            }
+            // if the total stake for this quorum was updated, record it in storage
+            if (totalStakeUpdate.updateBlockNumber != 0) {
+                // update the total stake history for the quorum
+                _recordTotalStakeUpdate(quorumNumber, totalStakeUpdate);
+            }
+            unchecked {
+                ++quorumNumber;
+            }
+        }
+
+        // record stake updates in the EigenLayer Slasher
+        for (uint i = 0; i < operators.length;) {
+            serviceManager.recordStakeUpdate(operators[i], uint32(block.number), serviceManager.latestServeUntilBlock(), prevElements[i]);
+            unchecked {
+                ++i;
+            }
+        }     
+    }
+
+    // INTERNAL FUNCTIONS
+
+    function _registerOperator(address operator, bytes32 operatorId, bytes memory quorumNumbers) internal {
+        uint8 quorumNumbersLength = uint8(quorumNumbers.length);
+        // check the operator is registering for only valid quorums
+        require(uint8(quorumNumbers[quorumNumbersLength - 1]) < quorumCount, "StakeRegistry._registerStake: greatest quorumNumber must be less than quorumCount");
+        OperatorStakeUpdate memory _newTotalStakeUpdate;
+        // add the `updateBlockNumber` info
+        _newTotalStakeUpdate.updateBlockNumber = uint32(block.number);
+        // for each quorum, evaluate stake and add to total stake
+        for (uint8 quorumNumbersIndex = 0; quorumNumbersIndex < quorumNumbersLength;) {
+            // get the next quourumNumber
+            uint8 quorumNumber = uint8(quorumNumbers[quorumNumbersIndex]);
+            // evaluate the stake for the operator
+            // since we don't use the first output, this will use 1 extra sload when deregistered operator's register again
+            (, uint96 stake) = _updateOperatorStake(operator, operatorId, quorumNumber);
+            // @JEFF: This reverts pretty late, but i think that's fine. wdyt?
+            // check if minimum requirement has been met, will be 0 if not
+            require(stake != 0, "StakeRegistry._registerStake: Operator does not meet minimum stake requirement for quorum");
+            // add operator stakes to total stake before update (in memory)
+            uint256 _totalStakeHistoryLength = _totalStakeHistory[quorumNumber].length;
+            if (_totalStakeHistoryLength != 0) {
+                // only add the stake if there is a previous total stake
+                // overwrite `stake` variable
+                stake += _totalStakeHistory[quorumNumber][_totalStakeHistoryLength - 1].stake;
+            }
+            _newTotalStakeUpdate.stake = stake;
+            // update storage of total stake
+            _recordTotalStakeUpdate(quorumNumber, _newTotalStakeUpdate);
+            unchecked {
+                ++quorumNumbersIndex;
+            }
+        }
+    }
+
+    /**
+     * @notice Removes the stakes of the operator
```

Comment:
comment feels a little lazy. is there any property(s) this should exhibit or more specific thing this is trying to accomplish?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/2631769ce9169012dc2edc79ac9c7834381de6d6